Update template with necessary files

TarasSpashchenko opened the pull request:
folio-spring-base dependency version bumped up to 4.1.0

update template with necessary files:
* main spring boot class
* basic test class
* Dockerfile
* replace application .properties-files with .yaml-files

TarasSpashchenko committed:
folio-spring-base version bump up to 4.1.0

TarasSpashchenko committed:
folio-spring-base version bump up to 4.1.0

julianladisch commented on pom.xml:
```diff
@@ -24,7 +24,7 @@
   <properties>
     <java.version>11</java.version>
-    <folio-spring-base.version>4.0.0</folio-spring-base.version>
+    <folio-spring-base.version>4.1.0</folio-spring-base.version>
```
folio-spring-base 4.1.0 is based on spring-boot-starter-parent 2.7, therefore spring-boot-starter-parent should be bumped from 2.6.4 to 2.7.1

TarasSpashchenko replied:
Thank you. Fixed.

TarasSpashchenko committed:
bump up spring-boot-starter-parent version to 2.7.1 to be consistent with folio-spring-base v 4.1.0

TarasSpashchenko committed:
bump up spring-boot-starter-parent version to 2.7.1 to be consistent with folio-spring-base v 4.1.0

TarasSpashchenko committed:
mod-spring-template readme file.

dmtkachenko committed:
update pom.xml
update application property files

dmtkachenko committed:
update pom.xml
update application property files

julianladisch commented on pom.xml:
```diff
@@ -35,6 +46,33 @@
       <artifactId>folio-spring-base</artifactId>
       <version>${folio-spring-base.version}</version>
     </dependency>
+	<dependency>
```
indentation

TarasSpashchenko replied:
Done

dmtkachenko committed:
basic classes for application and tests

dmtkachenko committed:
basic classes for application and tests

dmtkachenko committed:
fix indent

dmtkachenko committed:
fix indent

dmtkachenko committed:
add sonar exclusion

dmtkachenko committed:
add sonar exclusion

dmtkachenko committed:
update jar file name

dmtkachenko committed:
update jar file name

TarasSpashchenko committed:
mod-spring-template minor fixes

TarasSpashchenko committed:
mod-spring-template minor fixes

TarasSpashchenko committed:
DeploymentDescriptor-template.json: fixed @artifactId@ instead of module name
Jenkinsfile: buildNode bumped up to jenkins-agent-java17
pom.xml: Java version bumped up to Java 17

TarasSpashchenko committed:
DeploymentDescriptor-template.json: fixed @artifactId@ instead of module name
Jenkinsfile: buildNode bumped up to jenkins-agent-java17
pom.xml: Java version bumped up to Java 17

TarasSpashchenko committed:
Type fixes.
Dockerfile removed because the correct one will be added in another branch.

dcrossleyau commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM folioci/alpine-jre-openjdk17:latest
+
+# Copy your fat jar to the container provide the actual name for your fat jar file for example mod-notes-fat.jar
+ENV APP_FILE mod-spring-tempale.jar
```
More typos

dmtkachenko replied:
updated

TarasSpashchenko committed:
Copyright fixed

TarasSpashchenko committed:
Merge pull request #9 from folio-org/feature/add-how-to

mod-spring-template readme file.

TarasSpashchenko committed:
Merge branch 'feature/folio-spring-base-version-bumpup-4.1.0' of github.com:folio-org/mod-spring-template into feature/folio-spring-base-version-bumpup-4.1.0

dmtkachenko committed:
typo fix
